tests/components/homekit/test_type_humidifiers.py

```diff
@@ -0,0 +1,419 @@
+        entity_id,
+        1,
+        {CONF_LINKED_HUMIDITY_SENSOR: humidity_sensor_entity_id},
+    )
+    hk_driver.add_accessory(acc)
+
+    await acc.run_handler()
+    await hass.async_block_till_done()
+
+    assert acc.char_current_humidity.value == 42.0
+
+    hass.states.async_set(
+        humidity_sensor_entity_id,
+        "43.0",
+        {
+            ATTR_DEVICE_CLASS: DEVICE_CLASS_HUMIDITY,
+            ATTR_UNIT_OF_MEASUREMENT: UNIT_PERCENTAGE,
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert acc.char_current_humidity.value == 43.0
+
+    hass.states.async_set(
+        humidity_sensor_entity_id,
+        STATE_UNAVAILABLE,
+        {
+            ATTR_DEVICE_CLASS: DEVICE_CLASS_HUMIDITY,
+            ATTR_UNIT_OF_MEASUREMENT: UNIT_PERCENTAGE,
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert acc.char_current_humidity.value == 43.0
+
+    hass.states.async_remove(humidity_sensor_entity_id)
+    await hass.async_block_till_done()
+
+    assert acc.char_current_humidity.value == 43.0
+
+
+async def test_humidifier_with_a_missing_linked_humidity_sensor(hass, hk_driver):
+    """Test a humidifier with a configured linked motion sensor that is missing."""
+    humidity_sensor_entity_id = "sensor.bedroom_humidity"
+    entity_id = "humidifier.test"
+
+    hass.states.async_set(entity_id, STATE_OFF)
+    await hass.async_block_till_done()
+    acc = HumidifierDehumidifier(
+        hass,
+        hk_driver,
+        "HumidifierDehumidifier",
+        entity_id,
+        1,
+        {CONF_LINKED_HUMIDITY_SENSOR: humidity_sensor_entity_id},
+    )
+    hk_driver.add_accessory(acc)
+
+    await acc.run_handler()
+    await hass.async_block_till_done()
+
+    assert acc.char_current_humidity.value == 0
+
+
+async def test_humidifier_as_dehumidifier(hass, hk_driver, events, caplog):
+    """Test an invalid char_target_humidifier_dehumidifier from HomeKit."""
+    entity_id = "humidifier.test"
+
+    hass.states.async_set(entity_id, STATE_OFF)
+    await hass.async_block_till_done()
+    acc = HumidifierDehumidifier(
+        hass, hk_driver, "HumidifierDehumidifier", entity_id, 1, None
+    )
+    hk_driver.add_accessory(acc)
+
+    await acc.run_handler()
+    await hass.async_block_till_done()
+
+    assert acc.char_target_humidifier_dehumidifier.value == 1
+
+    # Set from HomeKit
+    char_target_humidifier_dehumidifier_iid = acc.char_target_humidifier_dehumidifier.to_HAP()[
+        HAP_REPR_IID
+    ]
+
+    hk_driver.set_characteristics(
+        {
+            HAP_REPR_CHARS: [
+                {
+                    HAP_REPR_AID: acc.aid,
+                    HAP_REPR_IID: char_target_humidifier_dehumidifier_iid,
+                    HAP_REPR_VALUE: 0,
+                },
+            ]
+        },
+        "mock_addr",
+    )
+
+    await hass.async_block_till_done()
+    assert "TargetHumidifierDehumidifierState is not supported" in caplog.text
+    assert len(events) == 0
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5795327)